### PR TITLE
added some module streams ui tests

### PR DESCRIPTION
**Pull request by omkarkhatavkar**
**Objective** 

Add UI test coverage for module streams 

**Test Result**  

```
(env) root@okhatavk robottelo (module_streams_ui_tests) $ pytest tests/foreman/ui_airgun/ -v -k 'module_stream' 
============================================================================ test session starts ============================================================================
platform linux -- Python 3.4.8, pytest-4.0.1, py-1.7.0, pluggy-0.8.0 -- /home/okhatavk/Satellite/robottelo/env/bin/python3.4
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/okhatavk/Satellite/robottelo, inifile:
plugins: xdist-1.24.1, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.0
collecting 189 items                                                                                                                                                        2019-01-03 15:24:00 - conftest - DEBUG - BZ deselect is disabled in settings

collected 268 items / 264 deselected                                                                                                                                        

tests/foreman/ui_airgun/test_contentview.py::test_positive_module_stream_end_to_end PASSED                                                                            [ 25%]
tests/foreman/ui_airgun/test_contentview.py::test_positive_search_module_streams_in_content_view PASSED                                                               [ 50%]
tests/foreman/ui_airgun/test_lifecycleenvironment.py::test_positive_search_lce_content_view_module_streams_by_name PASSED                                             [ 75%]
tests/foreman/ui_airgun/test_modulestreams.py::test_positive_module_stream_details_search_in_repo PASSED                                                              [100%]

============================================================================= warnings summary ==============================================================================
tests/foreman/ui_airgun/test_lifecycleenvironment.py::test_positive_search_lce_content_view_module_streams_by_name
  /home/okhatavk/Satellite/robottelo/env/lib/python3.4/site-packages/widgetastic/widget/select.py:130: DeprecationWarning: The value of convert_charrefs will become True in 3.5. You are encouraged to set the value explicitly.
    parser = html_parser.HTMLParser()
  /home/okhatavk/Satellite/robottelo/env/lib/python3.4/site-packages/widgetastic/widget/select.py:133: DeprecationWarning: The unescape method is deprecated and will be removed in 3.5, use html.unescape() instead.
    for option
  /home/okhatavk/Satellite/robottelo/env/lib/python3.4/site-packages/widgetastic/widget/select.py:130: DeprecationWarning: The value of convert_charrefs will become True in 3.5. You are encouraged to set the value explicitly.
    parser = html_parser.HTMLParser()
  /home/okhatavk/Satellite/robottelo/env/lib/python3.4/site-packages/widgetastic/widget/select.py:133: DeprecationWarning: The unescape method is deprecated and will be removed in 3.5, use html.unescape() instead.
    for option

-- Docs: https://docs.pytest.org/en/latest/warnings.html
========================================================== 4 passed, 264 deselected, 4 warnings in 721.67 seconds ===========================================================
(env) root@okhatavk robottelo (module_streams_ui_tests) $ 
```

**Dependent Airgun PR** 
https://github.com/SatelliteQE/airgun/pull/270

### CODE REVIEW

**Comment by oshtaier** [comment]
From application perspective, module stream is separate entity, so test cases related to that specific page should be placed into separate file.
If we validate that everything okay for content view (and use module stream for it) then we should place such test case into content_view.py and so forth

File: tests/foreman/ui_airgun/test_contentview.py [hunk 1]
```diff
@@ -2058,3 +2059,92 @@ def test_positive_composite_child_inc_update(session):
...
+    :expectedresults: Content view works properly with module_streams and
+        count shown should be correct
+
+    :CaseLevel: System
+    """
+    repo_name = gen_string('alpha')
+    env_name = gen_string('alpha')
+    cv_name = gen_string('alpha')
+    # Creates a CV along with product and sync'ed repository
+    create_sync_custom_repo(module_org.id, repo_name=repo_name,
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_contentview.py]:
```
create_sync_custom_repo(
    module_org.id, 
    repo_name=repo_name,
    repo_url=CUSTOM_MODULE_STREAM_REPO_2)
```

Comment (oshtaier) [reply tests/foreman/ui_airgun/test_contentview.py]: Second: 
Is module stream repo has `REPO_TYPE['yum']` type? Or it has its own unique type?

File: tests/foreman/ui_airgun/test_contentview.py [hunk 2]
```diff
@@ -2058,3 +2059,92 @@ def test_positive_composite_child_inc_update(session):
...
+    :id: 7f5273ff-e80f-459d-adf4-b517b6d60fdc
+
+    :expectedresults: Searching for module streams should work inside content
+        view version
+
+    :CaseLevel: Integration
+    """
+    repo_name = gen_string('alpha')
+    module_stream = 'walrus'
+    create_sync_custom_repo(module_org.id, repo_name=repo_name,
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_contentview.py]:
```
create_sync_custom_repo(
    module_org.id, 
    repo_name=repo_name,
    repo_url=CUSTOM_MODULE_STREAM_REPO_2
)
```

File: tests/foreman/ui_airgun/test_contentview.py (hunk 1 above)
Comment (omkarkhatavkar) [reply tests/foreman/ui_airgun/test_contentview.py]: REPO_TYPE['yum'] is correct, module streams derived from yum repos only

File: tests/foreman/ui_airgun/test_contentview.py [hunk 3]
```diff
@@ -2058,3 +2059,92 @@ def test_positive_composite_child_inc_update(session):
...
+    repo = entities.Repository(name=repo_name).search(
+        query={'organization_id': module_org.id})[0]
+    cv = entities.ContentView(
+        organization=module_org,
+        repository=[repo]
+    ).create()
+    with session:
+        result = session.contentview.publish(cv.name)
+        assert result['Version'] == VERSION
+        module_streams = session.contentview.search_version_module_stream(
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_contentview.py]:
```
        module_streams = session.contentview.search_version_module_stream(
            cv.name, 
            VERSION,
            'name = "{}" and stream = "{}"'.format(module_stream, '0.71')
        )
```

File: tests/foreman/ui_airgun/test_contentview.py [hunk 4]
```diff
@@ -2058,3 +2059,92 @@ def test_positive_composite_child_inc_update(session):
...
+            cv.name, VERSION,
+            'name = "{}" and stream = "{}"'.format(module_stream, '0.71')
+        )
+        assert len(module_streams) == 1
+        assert (
+            module_streams[0]['Name'] == module_stream
+            and module_streams[0]['Stream'] == '0.71'
+        )
+        module_streams = session.contentview.search_version_module_stream(
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_contentview.py]:
```
        module_streams = session.contentview.search_version_module_stream(
            cv.name, 
            VERSION,
            'name = "{}" and stream = "{}"'.format(module_stream, '5.21')
        )
```

File: tests/foreman/ui_airgun/test_contentview.py (hunk 3 above)
Comment (omkarkhatavkar) [reply tests/foreman/ui_airgun/test_contentview.py]: Nice! work on it

File: tests/foreman/ui_airgun/test_lifecycleenvironment.py [hunk 5]
```diff
@@ -254,6 +256,54 @@ def test_positive_search_lce_content_view_packages_by_name(
...
+            - P1 has two module streams
+            - P2 has three module streams
+        2. Create a content view with the repository and publish it
+        3. Go to Lifecycle Environment > Library > ModuleStreams
+        4. Select the content view
+        5. Search by module stream names
+
+    :expectedresults: only the searched module streams where found
+
+    :BZ: 1432155
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_lifecycleenvironment.py]: 1432155 talks about packages, not about module streams, so I would remove:
```
:BZ: 1432155
```
and
```
:customerscenario: true
```

File: tests/foreman/ui_airgun/test_lifecycleenvironment.py [hunk 6]
```diff
@@ -254,6 +256,54 @@ def test_positive_search_lce_content_view_packages_by_name(
...
+
+    :CaseLevel: System
+    """
+    module_streams = [
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_lifecycleenvironment.py]:
```
    module_streams = [
        {
             'name': FAKE_1_CUSTOM_PACKAGE_NAME,
             'streams_count': 2
        },
        {
             'name': FAKE_3_CUSTOM_PACKAGE_NAME,
             'streams_count': 3
        },
    ]
```

File: tests/foreman/ui_airgun/test_lifecycleenvironment.py (hunk 5 above)
Comment (omkarkhatavkar) [reply tests/foreman/ui_airgun/test_lifecycleenvironment.py]: yeah copy+paste mistake

**Comment by JacobCallahan** [comment]
@oshtaier I agree with your comment above since I wasn't aware the module streams are a top-level feature in 6.5.

File: tests/foreman/ui_airgun/test_contentview.py (hunk 3 above)
Comment (omkarkhatavkar) [reply tests/foreman/ui_airgun/test_contentview.py]: This is done @oshtaier

**Comment by omkarkhatavkar** [comment]
@oshtaier : I have posted the result here https://github.com/SatelliteQE/airgun/pull/270#issuecomment-454756729, can we please approve this merge request if all looks good?

File: tests/foreman/ui_airgun/test_contentview.py [hunk 7]
```diff
@@ -2058,3 +2059,98 @@ def test_positive_composite_child_inc_update(session):
...
+    repo = entities.Repository(name=repo_name).search(
+        query={'organization_id': module_org.id})[0]
+    cv = entities.ContentView(
+        organization=module_org,
+        repository=[repo]
+    ).create()
+    with session:
+        result = session.contentview.publish(cv.name)
+        assert result['Version'] == VERSION
+        module_streams = session.contentview.search_version_module_stream(
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_contentview.py]:
```
for module_version in ['0.71', '5.21']:
    module_streams = session.contentview.search_version_module_stream(
            cv.name,
            VERSION,
            'name = "{}" and stream = "{}"'.format(module_stream, module_version )
        )
        assert len(module_streams) == 1
        assert (
            module_streams[0]['Name'] == module_stream
            and module_streams[0]['Stream'] == module_version 
        )

File: tests/foreman/ui_airgun/test_contentview.py [hunk 8]
```diff
@@ -2058,3 +2059,98 @@ def test_positive_composite_child_inc_update(session):
...
+    :id: 7f5273ff-e80f-459d-adf4-b517b6d60fdc
+
+    :expectedresults: Searching for module streams should work inside content
+        view version
+
+    :CaseLevel: Integration
+    """
+    repo_name = gen_string('alpha')
+    module_stream = 'walrus'
+    create_sync_custom_repo(module_org.id,
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_contentview.py]: nitpick:
```
create_sync_custom_repo(
    module_org.id,
    repo_name=repo_name,
    repo_url=CUSTOM_MODULE_STREAM_REPO_2
)
```

File: tests/foreman/ui_airgun/test_modulestreams.py [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        walrus_details = session.modulestream.read('walrus')
+        expected_module_details = {
+            'Summary': 'Walrus 5.21 module',
+            'Context': 'deadbeef',
+            'Name': 'walrus',
+            'Stream': '5.21',
+            'Arch': 'x86_64',
+            'Description': 'A module for the walrus 5.21 package',
+        }
+        print("walrus_details ======> {}".format(walrus_details))
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_modulestreams.py]: ! remove print

File: tests/foreman/ui_airgun/test_contentview.py (hunk 7 above)
Comment (omkarkhatavkar) [reply tests/foreman/ui_airgun/test_contentview.py]: done !

Comment (omkarkhatavkar) [reply tests/foreman/ui_airgun/test_contentview.py]: done !

File: tests/foreman/ui_airgun/test_contentview.py (hunk 8 above)
Comment (omkarkhatavkar) [reply tests/foreman/ui_airgun/test_contentview.py]: done !

File: tests/foreman/ui_airgun/test_modulestreams.py (hunk 9 above)
Comment (omkarkhatavkar) [reply tests/foreman/ui_airgun/test_modulestreams.py]: done !

**Comment by oshtaier** [comment]
Omkar, please, once more, do fixes in separate commits as everyone do, so reviewers can look on the changes, not whole PR

File: tests/foreman/ui_airgun/test_contentview.py (hunk 8 above)
Comment (oshtaier) [reply tests/foreman/ui_airgun/test_contentview.py]: I don't see changes here. Did I miss something?

**Comment by omkarkhatavkar** [comment]
> Omkar, please, once more, do fixes in separate commits as everyone do, so reviewers can look on the changes, not whole PR

OK from next time I will make sure, I thought it was a very small change.

**Comment by oshtaier** [comment]
yeah-yeah, nvm, I just wanted to enforce approach as a standard, nothing more. I have no issues to review PR in that particular situation, we just do that even for 1 liner changes

**Comment by omkarkhatavkar** [comment]
> ACK
> pending test result for `test_positive_search_module_streams_in_content_view`

good to go now!

```
env) [root@okhatavk robottelo]# pytest tests/foreman/ui_airgun/ -v -k "test_positive_search_module_streams_in_content_view"
========================================================= test session starts ==========================================================
platform linux -- Python 3.4.8, pytest-4.0.2, py-1.7.0, pluggy-0.8.0 -- /home/okhatavk/Satellite/robottelo/env/bin/python3.4
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/okhatavk/Satellite/robottelo, inifile:
plugins: xdist-1.24.1, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.0
collecting 209 items                                                                                                                   2019-01-16 19:40:30 - conftest - DEBUG - BZ deselect is disabled in settings

collected 268 items / 267 deselected                                                                                                   

tests/foreman/ui_airgun/test_contentview.py::test_positive_search_module_streams_in_content_view PASSED                          [100%]

============================================== 1 passed, 267 deselected in 195.14 seconds ==============================================

```

**Comment by oshtaier** [comment]
Thanks for your patience =)

**Pull request merged**